### ISSUE TITLE
Is the `fiftyone.core.labels.Attribute` class doing anything?

### ISSUE THREAD

**Issue by tylerganter** [voxel51/fiftyone#253]
Brought up from a conversation with @ehofesmann 

https://github.com/voxel51/fiftyone/blob/488962848289986636a1fa6f92a95fea1a1b2530/fiftyone/core/labels.py#L69

This is not, to my knowledge being used by the app as something that demands "forced structure", so could we replace this with simple structures like dictionaries?

e.g.:

```python
my_detection["my_bool_attribute"] = True
my_detection["my_categorical_attribute"] = {
    "value": "tiger",
    "confidence": 0.9,
    "logits": [1, 2, 3, 4]
}
```
